// uu_plus4u5g01/dist-node/app/authors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Authors = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

var _uu_plus4u5g = require("uu_plus4u5g01");

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

require("uu_plus4u5g01-bricks");

require("uu5g04-bricks");

require("./authors.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Authors = exports.Authors = (0, _createReactClass2.default)({
  displayName: "Authors",

  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.SectionMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Authors',
    classNames: {
      main: 'plus4u5-app-authors',
      header: 'plus4u5-app-authors-header',
      authors: 'plus4u5-app-authors-authors'
    },
    opt: {
      dummyLevel: true
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    leadingAuthors: _propTypes2.default.arrayOf(_propTypes2.default.shape({
      name: _propTypes2.default.string.isRequired,
      content: _propTypes2.default.string,
      uuIdentity: _propTypes2.default.string,
      signature: _propTypes2.default.string,
      role: _propTypes2.default.any.isRequired
    })),
    otherAuthors: _propTypes2.default.arrayOf(_propTypes2.default.shape({
      name: _propTypes2.default.string,
      content: _propTypes2.default.string,
      uuIdentity: _propTypes2.default.string,
      role: _propTypes2.default.any
    }))
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      leadingAuthors: [],
      otherAuthors: null
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getAuthors: function _getAuthors(authors, size) {
    return authors.map(function (author, i) {
      return _react2.default.createElement(Author, _extends({ key: i }, author, { size: size }));
    });
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return _react2.default.createElement(
      UU5.Bricks.Section,
      _extends({}, this.getMainPropsToPass(), { content: null, level: this.props.level || 3 }),
      this.props.leadingAuthors && _react2.default.createElement(UU5.Bricks.Div, { className: this.getClassName().authors, content: this._getAuthors(this.props.leadingAuthors, 'm') }),
      this.props.otherAuthors && _react2.default.createElement(UU5.Bricks.Div, { className: this.getClassName().authors, content: this._getAuthors(this.props.otherAuthors, 's') })
    );
  }
  //@@viewOff:render

});

exports.default = Authors;


var Author = (0, _createReactClass2.default)({
  displayName: "Author",

  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.ContentMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Author',
    classNames: {
      main: 'plus4u5-app-author',
      size: 'plus4u5-app-author-size',
      image: 'plus4u5-app-author-image',
      signature: 'plus4u5-app-author-signature',
      name: 'plus4u5-app-author-name',
      position: 'plus4u5-app-author-position',
      content: 'plus4u5-app-author-content'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    signature: _propTypes2.default.string,
    src: _propTypes2.default.string,
    uuIdentity: _propTypes2.default.string,
    role: _propTypes2.default.any,
    size: _propTypes2.default.oneOf(['s', 'm'])
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      signature: null,
      src: null,
      uuIdentity: null,
      role: null,
      size: 'm'
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getMainAttrs: function _getMainAttrs() {
    var attrs = this.getMainPropsToPass();
    attrs.content = null;
    attrs.className += " " + this.getClassName().size + "-" + (this.props.content ? 'l' : this.props.size);
    return attrs;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return _react2.default.createElement(
      UU5.Bricks.Column,
      this._getMainAttrs(),
      _react2.default.createElement(Plus4U5.Bricks.UserPhoto, { className: this.getClassName().image, type: "circle", width: null, uuIdentity: this.props.uuIdentity, src: this.props.src }),
      this.props.signature && _react2.default.createElement(UU5.Bricks.Image, { className: this.getClassName().signature, src: this.props.signature }),
      _react2.default.createElement(UU5.Bricks.Span, { className: this.getClassName().name, content: this.props.name }),
      _react2.default.createElement(UU5.Bricks.Span, { className: this.getClassName().position, content: this.props.role }),
      this.props.content && _react2.default.createElement(UU5.Bricks.P, { className: this.getClassName().content, content: this.props.content })
    );
  }
  //@@viewOff:render

});
