// uu_plus4u5g01/dist-node/app/button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

var _navigator = require("./navigator.js");

var _navigator2 = _interopRequireDefault(_navigator);

require("./button.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Button = exports.Button = (0, _createReactClass2.default)({
  displayName: "Button",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.PureRenderMixin, UU5.Common.NestingLevelMixin, UU5.Common.IdentityMixin, UU5.Common.CcrReaderMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Button',
    nestingLevel: 'smallBox',
    classNames: {
      main: 'plus4u5-app-button uu5-noindex',
      authenticated: 'plus4u5-app-button-authenticated',
      notAuthenticated: 'plus4u5-app-button-not-authenticated',
      pending: 'plus4u5-app-button-pending'
    },
    defaults: {
      plus4uIcon: 'plus4u-glyphicon-plus4u'
    },
    opt: {
      nestingLevelWrapper: true
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    smallNav: _propTypes2.default.bool
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      smallNav: false
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getPropsToPass: function _getPropsToPass(mode) {
    var props = this.getMainPropsToPass();
    props = UU5.Common.Tools.merge(props, { className: props.className + ' ' + this.getClassName(mode) });
    return props;
  },
  _getChild: function _getChild() {
    var _this = this;

    var child = void 0;
    var icon = _react2.default.createElement(UU5.Bricks.Icon, { icon: this.getDefault().plus4uIcon });
    switch (this.getIdentityFeedback()) {
      case 'authenticated':
        child = _react2.default.createElement(
          "span",
          null,
          _react2.default.createElement(
            UU5.Bricks.Button,
            _extends({}, this._getPropsToPass('authenticated'), { colorSchema: "custom", onClick: function onClick() {
                return _this._navigator.toggleHidden();
              } }),
            icon
          ),
          _react2.default.createElement(_navigator2.default, { parent: this, uuIdentity: this.getIdentity().uuIdentity, hidden: true, small: this.props.smallNav, ref_: function ref_(nav) {
              return _this._navigator = nav;
            } })
        );
        break;
      case 'notAuthenticated':
        child = _react2.default.createElement(
          UU5.Bricks.Button,
          _extends({}, this._getPropsToPass('notAuthenticated'), { colorSchema: "custom", onClick: function onClick() {
              return _this.login();
            } }),
          icon
        );
        break;
      default:
        child = _react2.default.createElement(
          UU5.Bricks.Button,
          _extends({}, this._getPropsToPass('pending'), { colorSchema: "custom", disabled: true }),
          icon
        );
    }
    return child;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return this.getNestingLevel() ? this._getChild() : null;
  }
} //@@viewOff:render

);

exports.default = Button;
