// uu_plus4u5g01/dist-node/app/left.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Left = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./left.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Left = exports.Left = (0, _createReactClass2.default)({
  displayName: "Left",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Left',
    classNames: {
      main: 'plus4u5-app-left'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return _react2.default.createElement(
      UU5.Bricks.Div,
      this.getMainPropsToPass(),
      "Plus4U5.App.Left TODO ...",
      _react2.default.createElement(
        UU5.Bricks.Ol,
        null,
        _react2.default.createElement(
          UU5.Bricks.Li,
          null,
          _react2.default.createElement(UU5.Bricks.LinkUnicorn, null)
        ),
        _react2.default.createElement(
          UU5.Bricks.Li,
          null,
          _react2.default.createElement(UU5.Bricks.LinkPlus4U, null)
        ),
        _react2.default.createElement(
          UU5.Bricks.Li,
          null,
          _react2.default.createElement(UU5.Bricks.LinkUnicornCollege, null)
        ),
        _react2.default.createElement(
          UU5.Bricks.Li,
          null,
          _react2.default.createElement(UU5.Bricks.LinkUAF, null)
        ),
        _react2.default.createElement(
          UU5.Bricks.Li,
          null,
          _react2.default.createElement(UU5.Bricks.LinkUU5, null)
        )
      )
    );
  }
} //@@viewOff:render

);

exports.default = Left;
