// uu_plus4u5g01/dist-node/app/licence.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Licence = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./licence.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Licence = exports.Licence = (0, _createReactClass2.default)({
  displayName: "Licence",

  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.SectionMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Licence',
    classNames: {
      main: 'plus4u5-app-licence',
      header: 'plus4u5-app-licence-header',
      row: 'plus4u5-app-licence-row',
      rowTitle: 'plus4u5-app-licence-row-title',
      rowContent: 'plus4u5-app-licence-row-content'
    },
    lsi: {
      header: {
        cs: 'Vlastník licence',
        en: 'License'
      },
      organisation: {
        cs: 'Organizace',
        en: 'Organisation'
      },
      authorities: {
        cs: 'Kompetentní osoby',
        en: 'Authorities'
      }
    },
    opt: {
      dummyLevel: true
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    organisation: _propTypes2.default.shape({
      name: _propTypes2.default.string,
      uri: _propTypes2.default.string
    }),
    authorities: _propTypes2.default.arrayOf(_propTypes2.default.shape({
      name: _propTypes2.default.string,
      uri: _propTypes2.default.string
    }))
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      organisation: null,
      authorities: []
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getLinks: function _getLinks(links) {
    if (!links) return;

    if (Array.isArray(links)) {
      var retArr = [];
      links.forEach(function (link, i) {
        retArr.push(_react2.default.createElement(UU5.Bricks.Link, { key: i, href: link.uri, content: link.name, target: "_blank" }));
        if (i < links.length - 1) retArr.push(_react2.default.createElement(
          "span",
          null,
          ",\xA0"
        ));
      });
      return retArr;
    } else {
      return _react2.default.createElement(UU5.Bricks.Link, { href: links.uri, content: links.name, target: "_blank" });
    }
  },
  _getRow: function _getRow(title, content) {
    return _react2.default.createElement(
      UU5.Bricks.Div,
      { className: this.getClassName().row },
      _react2.default.createElement(UU5.Bricks.Span, { className: this.getClassName().rowTitle, content: title }),
      _react2.default.createElement(UU5.Bricks.Span, { className: this.getClassName().rowContent, content: this._getLinks(content) })
    );
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var header = this.props.header || this.getLsiComponent('header');

    return _react2.default.createElement(
      UU5.Bricks.Section,
      _extends({}, this.getMainPropsToPass(), { content: null, header: header, level: this.props.level || 4 }),
      this._getRow(this.getLsiComponent('organisation'), this.props.organisation),
      this._getRow(this.getLsiComponent('authorities'), this.props.authorities)
    );
  }
  //@@viewOff:render

});

exports.default = Licence;
