// uu_plus4u5g01/dist-node/app/loading.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Loading = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./loading.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Loading = exports.Loading = (0, _createReactClass2.default)({
  displayName: "Loading",

  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.ContentMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Loading',
    classNames: {
      main: 'plus4u5-app-loading uu5-noindex',
      spinner: 'plus4u5-app-loading-spinner',
      content: 'plus4u5-app-loading-content'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var content = this.getContent() || this.props.children || _react2.default.createElement(UU5.Bricks.Icon, { icon: "plus4u5-plus4u" });

    return _react2.default.createElement(
      UU5.Bricks.Loading,
      _extends({}, this.getMainPropsToPass(), { content: null }),
      _react2.default.createElement(UU5.Bricks.Div, { className: this.getClassName().spinner }),
      _react2.default.createElement(UU5.Bricks.Div, { className: this.getClassName().content, content: content })
    );
  }
  //@@viewOff:render

});

exports.default = Loading;
