// uu_plus4u5g01/dist-node/app/login-button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LoginButton = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LoginButton = exports.LoginButton = (0, _createReactClass2.default)({
  displayName: "LoginButton",


  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.PureRenderMixin, UU5.Common.ColorSchemaMixin, UU5.Common.IdentityMixin],

  statics: {
    tagName: 'Plus4U5.App.LoginButton',
    classNames: {
      main: 'plus4u5-app-login-button uu5-noindex'
    },
    lsi: {
      login: {
        cs: 'Přihlásit',
        en: 'Login'
      },
      logout: {
        cs: 'Odhlásit',
        en: 'Logout'
      }
    }
  },

  propTypes: {
    size: _propTypes2.default.string,
    displayBlock: _propTypes2.default.bool,
    pressed: _propTypes2.default.bool
  },

  getDefaultProps: function getDefaultProps() {
    return {
      size: 'm',
      displayBlock: false,
      pressed: false
    };
  },
  render: function render() {
    var _this = this;

    return _react2.default.createElement(UU5.Bricks.ButtonSwitch, _extends({}, this.getMainPropsToPass(), { colorSchema: this.props.colorSchema, size: this.props.size, props: {
        displayBlock: this.props.displayBlock,
        pressed: this.props.pressed
      }, switchedOn: this.isAuthenticated(), onProps: { content: this.getLsiComponent('logout'), onClick: function onClick() {
          return _this.logout();
        } }, offProps: { content: this.getLsiComponent('login'), onClick: function onClick() {
          return _this.login();
        } } }));
  }
});

exports.default = LoginButton;
