// uu_plus4u5g01/dist-node/app/login.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Login = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

var _uu_plus4u5g = require("uu_plus4u5g01");

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

var _loginButton = require("./login-button.js");

var _loginButton2 = _interopRequireDefault(_loginButton);

require("./login.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Login = exports.Login = (0, _createReactClass2.default)({
  displayName: "Login",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin, UU5.Common.LsiMixin, UU5.Common.IdentityMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Login',
    nestingLevelList: UU5.Environment.getNestingLevelList('box', 'inline'),
    classNames: {
      main: 'plus4u5-app-login uu5-noindex',
      text: 'plus4u5-app-login-text',
      buttons: 'plus4u5-app-login-buttons',
      loginText: 'plus4u5-app-login-login-text',
      registerText: 'plus4u5-app-login-register-text',
      loginButton: 'plus4u5-app-login-login-button',
      registerButton: 'plus4u5-app-login-register-button'
    },
    opt: {
      nestingLevelWrapper: true
    },
    lsi: {
      loginText: {
        cs: 'Nejprve se prosím přihlaste',
        en: 'To read the book, please log in…'
      },
      registerText: {
        cs: 'Pokud nemáte založen účet, můžete se zaregistrovat.',
        en: 'If you don’t have an account yet, feel free to make a simple registration.'
      },
      registerButton: {
        cs: 'Vytvořit účet',
        en: 'Create account'
      }
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _register: function _register() {
    var url = this.getLsiItem(Plus4U5.Environment.registrationUrl);
    window.open(url, '_blank');
    return this;
  },


  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return this.getNestingLevel() && this.isNotAuthenticated() ? _react2.default.createElement(
      UU5.Bricks.Div,
      this.getMainPropsToPass(),
      _react2.default.createElement(
        UU5.Bricks.Div,
        { className: this.getClassName('text') },
        _react2.default.createElement(UU5.Bricks.P, { className: this.getClassName('loginText'), content: this.getLsiValue('loginText') }),
        _react2.default.createElement(UU5.Bricks.P, { className: this.getClassName('registerText'), content: this.getLsiValue('registerText') })
      ),
      _react2.default.createElement(
        UU5.Bricks.Div,
        { className: this.getClassName('buttons') },
        _react2.default.createElement(_loginButton2.default, { className: this.getClassName('loginButton') }),
        _react2.default.createElement(UU5.Bricks.Button, { className: this.getClassName('registerButton'), onClick: this._register, content: this.getLsiValue('registerButton') })
      )
    ) : null;
  }
  //@@viewOff:render

});

exports.default = Login;
