// uu_plus4u5g01/dist-node/app/navigation-bar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NavigationBar = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

var _navigationBarLoader = require("./navigation-bar-loader.js");

var _navigationBarLoader2 = _interopRequireDefault(_navigationBarLoader);

require("./navigation-bar.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NavigationBar = exports.NavigationBar = (0, _createReactClass2.default)({
  displayName: "NavigationBar",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin, UU5.Common.PureRenderMixin, UU5.Common.IdentityMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.NavigationBar',
    nestingLevelList: UU5.Environment.getNestingLevelList('bigBoxCollection', 'box'),
    classNames: {
      main: 'plus4u5-app-navigation-bar'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    mainEntityUri: _propTypes2.default.string.isRequired,
    navigationBarUri: _propTypes2.default.string,
    target: _propTypes2.default.oneOf(["auto", "_blank", "blank", "_top", "top", "_self", "self", "_parent", "parent"])
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      navigationBarUri: null,
      target: "_blank"
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  reload: function reload() {
    this._loader.reload();
    return this;
  },

  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _fixUesUri: function _fixUesUri(uri) {
    if (!uri) return uri;
    return uri.substring(0, 4) === 'ues:' ? uri : 'ues:' + uri;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var _this = this;

    var child = null;

    if (this.getNestingLevel() && this.isAuthenticated()) {
      child = _react2.default.createElement(_navigationBarLoader2.default, _extends({}, this.getMainPropsToPass(), { mainEntityUri: this._fixUesUri(this.props.mainEntityUri), navigationBarUri: this._fixUesUri(this.props.navigationBarUri), target: this.props.target, uuIdentityName: this.getIdentity().name, ref_: function ref_(loader) {
          return _this._loader = loader;
        } }));
    }

    return child;
  }
} //@@viewOff:render

);

exports.default = NavigationBar;
