// uu_plus4u5g01/dist-node/app/notification-badge-loader_.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

var _calls = require("./calls.js");

var _calls2 = _interopRequireDefault(_calls);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NotificationBadgeLoader = (0, _createReactClass2.default)({
  displayName: "NotificationBadgeLoader",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.PureRenderMixin, UU5.Common.NestingLevelMixin, UU5.Common.LoadMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.NotificationBadgeLoader',
    nestingLevel: 'smallBox',
    classNames: {
      main: 'plus4u5-app-notification-badge-loader'
    },
    calls: {
      onLoad: 'loadActivities',
      onReload: 'loadActivities'
    },
    opt: {
      nestingLevelWrapper: true,
      reloadInterval: 5 * 60 * 1000
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  componentWillMount: function componentWillMount() {
    this.setCalls(_calls2.default);
  },


  // componentWillReceiveProps() {
  //   Calls[this.getCalls.onLoad];
  // },
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  getOnLoadData_: function getOnLoadData_(props) {
    return {
      uuIdentity: props.uuIdentity
    };
  },

  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  onLoadSuccess_: function onLoadSuccess_(dtoOut) {
    // TODO: why? it should be some event when reload is called
    UU5.Environment.uu5DataMap['Plus4U5.App.NotificationBadge'] = {
      timeStamp: Date.now(),
      data: dtoOut
    };
    this.onLoadSuccess(dtoOut);
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var child = null;

    if (this.isReady()) {
      var count = 0;
      var dtoOut = this.getDtoOut();
      if (dtoOut && dtoOut.pageEntries && (count = dtoOut.pageEntries.length)) {
        child = _react2.default.createElement(UU5.Bricks.Badge, _extends({}, this.getMainPropsToPass(), { colorSchema: "danger", content: count }));
      }
    }

    return child;
  }
} //@@viewOff:render
);

exports.default = NotificationBadgeLoader;
