// uu_plus4u5g01/dist-node/app/page.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Page = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

var _top = require("./top.js");

var _top2 = _interopRequireDefault(_top);

var _bottom = require("./bottom.js");

var _bottom2 = _interopRequireDefault(_bottom);

var _button = require("./button.js");

var _button2 = _interopRequireDefault(_button);

var _notificationBadge = require("./notification-badge.js");

var _notificationBadge2 = _interopRequireDefault(_notificationBadge);

require("./page.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Page = exports.Page = (0, _createReactClass2.default)({
  displayName: "Page",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin, UU5.Common.ContentMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Page',
    nestingLevel: 'page',
    classNames: {
      main: 'plus4u5-app-page',
      topWrapper: 'plus4u5-app-page-top-wrapper',
      bottomWrapper: 'plus4u5-app-page-bottom-wrapper',
      leftWrapper: 'plus4u5-app-page-left-wrapper',
      rightWrapper: 'plus4u5-app-page-right-wrapper',
      contentWrapper: 'plus4u5-app-page-content-wrapper',
      systemLayerWrapper: 'plus4u5-app-page-system-layer-wrapper',
      appLayerWrapper: 'plus4u5-app-page-app-layer-wrapper',
      languageSelector: 'plus4u5-app-page-language-selector'
    },
    opt: {
      nestingLevelWrapper: true
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    displayedLanguages: _propTypes2.default.oneOfType([_propTypes2.default.arrayOf(_propTypes2.default.string), _propTypes2.default.string]),
    fullPage: _propTypes2.default.bool
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      displayedLanguages: null,
      fullPage: true
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  componentWillMount: function componentWillMount() {
    this.props.userLayerContent && UU5.Common.Tools.warning("Property 'userLayerContent' is deprecated! Use 'appLayerContent' instead.");
    this.props.userLayerWrapperProps && UU5.Common.Tools.warning("Property 'userLayerWrapperProps' is deprecated! Use 'appLayerWrapperProps' instead.");
  },

  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getLanguageSelector: function _getLanguageSelector() {
    return _react2.default.createElement(UU5.Bricks.LanguageSelector, { headerMode: "code", bodyMode: "label-code", noIndex: true, pullRight: true, displayedLanguages: this.props.displayedLanguages, className: this.getClassName('languageSelector') });
  },
  _expandWrapperProps: function _expandWrapperProps(props, className) {
    var wrapperProps = UU5.Common.Tools.mergeDeep({}, props) || {};
    wrapperProps.className = wrapperProps.className || '';
    wrapperProps.className += ' ' + this.getClassName(className);
    return wrapperProps;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var topWrapperProps = this._expandWrapperProps(this.props.topWrapperProps, 'topWrapper');
    var bottomWrapperProps = this._expandWrapperProps(this.props.bottomWrapperProps, 'bottomWrapper');
    var leftWrapperProps = this._expandWrapperProps(this.props.leftWrapperProps, 'leftWrapper');
    var rightWrapperProps = this._expandWrapperProps(this.props.rightWrapperProps, 'rightWrapper');
    var contentWrapperProps = this._expandWrapperProps(this.props.contentWrapperProps, 'contentWrapper');
    var appLayerWrapperProps = this._expandWrapperProps(this.props.appLayerWrapperProps || this.props.userLayerWrapperProps, 'appLayerWrapper');
    var systemLayerWrapperProps = this._expandWrapperProps(this.props.systemLayerWrapperProps, 'systemLayerWrapper');

    return this.getNestingLevel() === 'page' ? _react2.default.createElement(
      UU5.Bricks.Page,
      _extends({}, this.getMainPropsToPass(), { type: this.props.type || 1, fullPage: this.props.fullPage, topWrapperProps: topWrapperProps, bottomWrapperProps: bottomWrapperProps, leftWrapperProps: leftWrapperProps, rightWrapperProps: rightWrapperProps, contentWrapperProps: contentWrapperProps, appLayerWrapperProps: appLayerWrapperProps, systemLayerWrapperProps: systemLayerWrapperProps, top: this.props.top === undefined ? _react2.default.createElement(_top2.default, null) : this.props.top, bottom: this.props.bottom === undefined ? _react2.default.createElement(_bottom2.default, null) : this.props.bottom, left: this.props.left, right: this.props.right, leftOpen: this.props.leftOpen, leftClosed: this.props.leftClosed, rightOpen: this.props.rightOpen, rightClosed: this.props.rightClosed, leftWidth: this.props.leftWidth, rightWidth: this.props.rightWidth, alertBus: this.props.alertBus === undefined ? _react2.default.createElement(UU5.Bricks.AlertBus, null) : this.props.alertBus, modal: this.props.modal === undefined ? _react2.default.createElement(UU5.Bricks.Modal, null) : this.props.modal, systemLayerContent: this.props.systemLayerContent === undefined ? [this._getLanguageSelector(), _react2.default.createElement(_button2.default, null), _react2.default.createElement(_notificationBadge2.default, null), _react2.default.createElement(UU5.Bricks.ButtonToTop, null)] : this.props.systemLayerContent, appLayerContent: this.props.appLayerContent || this.props.userLayerContent }),
      this.props.children
    ) : null;
  }
} //@@viewOff:render

);

exports.default = Page;
