// uu_plus4u5g01/dist-node/app/resources.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Resources = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./resources.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Resources = exports.Resources = (0, _createReactClass2.default)({
  displayName: "Resources",

  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.SectionMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Resources',
    classNames: {
      main: 'plus4u5-app-resources',
      resources: 'plus4u5-app-resources-resources',
      resource: 'plus4u5-app-resources-resource'
    },
    lsi: {
      header: {
        cs: 'Použité zdroje',
        en: 'Resources'
      }
    },
    opt: {
      dummyLevel: true
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    resources: _propTypes2.default.array
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      resources: []
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getResources: function _getResources() {
    var _this = this;

    return this.props.resources.map(function (text, i) {
      return _react2.default.createElement(UU5.Bricks.Div, { className: _this.getClassName().resource, key: i, content: text });
    });
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var header = this.props.header ? this.props.header : this.getLsiComponent('header');

    return _react2.default.createElement(UU5.Bricks.Section, _extends({}, this.getMainPropsToPass(), { level: this.props.level || 4, header: header, content: this._getResources() }));
  }
  //@@viewOff:render

});

exports.default = Resources;
