// uu_plus4u5g01/dist-node/app/right.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Right = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./right.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Right = exports.Right = (0, _createReactClass2.default)({
  displayName: "Right",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Right',
    nestingLevel: 'boxCollection',
    classNames: {
      main: 'plus4u5-app-right'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return _react2.default.createElement(
      UU5.Bricks.Div,
      this.getMainPropsToPass(),
      _react2.default.createElement(UU5.Bricks.Button, { content: _react2.default.createElement(UU5.Bricks.Icon, { icon: "uu-glyphicon-calendar" }), colorSchema: "success" }),
      _react2.default.createElement(UU5.Bricks.Button, { content: _react2.default.createElement(UU5.Bricks.Icon, { icon: "glyphicon-envelope" }), colorSchema: "primary" }),
      _react2.default.createElement(UU5.Bricks.Button, { content: _react2.default.createElement(UU5.Bricks.Icon, { icon: "glyphicon-bell" }), colorSchema: "grey-rich" }),
      _react2.default.createElement(UU5.Bricks.Button, { content: _react2.default.createElement(UU5.Bricks.Icon, { icon: "glyphicon-wrench" }), colorSchema: "warning" })
    );
  }
} //@@viewOff:render

);

exports.default = Right;
