// uu_plus4u5g01/dist-node/app/search-button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchButton = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./search-button.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SearchButton = exports.SearchButton = (0, _createReactClass2.default)({
  displayName: "SearchButton",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin, UU5.Common.CcrReaderMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.SearchButton',
    nestingLevel: 'smallBox',
    classNames: {
      main: 'plus4u5-app-search-button'
    },
    opt: {
      nestingLevelWrapper: true
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _onClickHandler: function _onClickHandler() {
    var modal = this.getCcrComponentByKey(UU5.Environment.CCRKEY_PAGE).getModal();
    if (modal) {
      modal.open({
        header: _react2.default.createElement(UU5.Bricks.Lsi, { lsi: { cs: 'Hledat', en: 'Search' } }),
        content: 'TODO: ....',
        footer: 'TODO: ....',
        sticky: false
      });
    }
    return this;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return this.getNestingLevel() ? _react2.default.createElement(UU5.Bricks.Button, _extends({}, this.getMainPropsToPass(), { content: _react2.default.createElement(UU5.Bricks.Icon, { icon: "mdi-magnify" }), onClick: this._onClickHandler })) : null;
  }
} //@@viewOff:render

);

exports.default = SearchButton;
