// uu_plus4u5g01/dist-node/app/spa.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Spa = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

var _login = require("./login.js");

var _login2 = _interopRequireDefault(_login);

var _page = require("./page.js");

var _page2 = _interopRequireDefault(_page);

require("./spa.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Spa = exports.Spa = (0, _createReactClass2.default)({
  displayName: "Spa",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin, UU5.Common.IdentityMixin, UU5.Common.ContentMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Spa',
    nestingLevel: 'spa',
    classNames: {
      main: 'plus4u5-app-spa'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    notAuthenticatedContent: _propTypes2.default.any
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      notAuthenticatedContent: null
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return this.getNestingLevel() ? _react2.default.createElement(
      _page2.default,
      _extends({}, this.getMainPropsToPass(), { type: 1, name: this.isAuthenticated() ? "authenticated" : "notAuthenticated" }),
      this.isAuthenticated() ? _react2.default.Children.toArray(this.props.children) : this.props.notAuthenticatedContent || _react2.default.createElement(_login2.default, null)
    ) : null;
  }
  //@@viewOff:render

});

exports.default = Spa;
