// uu_plus4u5g01/dist-node/app/technologies.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Technologies = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./technologies.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Technologies = exports.Technologies = (0, _createReactClass2.default)({
  displayName: "Technologies",

  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.SectionMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Technologies',
    classNames: {
      main: 'plus4u5-app-technologies',
      mainTechnologies: 'plus4u5-app-technologies-main-technologies',
      row: 'plus4u5-app-technologies-row'
    },
    lsi: {
      header: {
        cs: 'Použité technologie',
        en: 'Used Technologies'
      },
      technologies: {
        cs: '<uu5string/>V aplikaci byly použity následující technologie společnosti <UU5.Bricks.Link href="http://www.unicorn.com/" content="Unicorn" target="_blank" />:',
        en: '<uu5string/>In the application were used following <UU5.Bricks.Link href="http://www.unicorn.com/" content="Unicorn" target="_blank" /> technologies:'
      },
      technologiesAnd: {
        cs: ' a ',
        en: ' and '
      }
    },
    opt: {
      dummyLevel: true
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    technologies: _propTypes2.default.array
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      technologies: []
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getTechnologies: function _getTechnologies() {
    var technologies = this.props.technologies;
    if (technologies.length > 1) technologies.splice(technologies.length - 1, 0, this.getLSIComponent("technologiesAnd"));
    var retArr = [];
    technologies.forEach(function (item, i) {
      retArr.push(item);
      if (i < technologies.length - 3) retArr.push(', ');else retArr.push(' ');
    });
    return retArr;
  },
  _getRows: function _getRows() {
    var _this = this;

    if (!this.props.content) return;

    if (Array.isArray(this.props.content)) {
      return this.props.content.map(function (row, i) {
        return _react2.default.createElement(UU5.Bricks.P, { className: _this.getClassName().row, key: i, content: row });
      });
    } else {
      return _react2.default.createElement(UU5.Bricks.P, { className: this.getClassName().row, content: this.props.content });
    }
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var header = this.props.header ? this.props.header : this.getLsiComponent('header');

    return _react2.default.createElement(
      UU5.Bricks.Section,
      _extends({}, this.getMainPropsToPass(), { header: header, content: null, level: this.props.level || 4 }),
      _react2.default.createElement(
        UU5.Bricks.Div,
        { className: this.getClassName().mainTechnologies },
        _react2.default.createElement(
          UU5.Bricks.Span,
          null,
          this.getLsiComponent('technologies')
        ),
        _react2.default.createElement(UU5.Bricks.Span, { content: this._getTechnologies() })
      ),
      this._getRows()
    );
  }
  //@@viewOff:render

});

exports.default = Technologies;
