// uu_plus4u5g01/dist-node/app/top.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Top = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("./top.less");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Top = exports.Top = (0, _createReactClass2.default)({
  displayName: "Top",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin, UU5.Common.CcrReaderMixin, UU5.Common.ContentMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.App.Top',
    nestingLevel: 'box',
    classNames: {
      main: 'plus4u5-app-top uu5-noindex',
      leftMenu: 'plus4u5-app-top-left-menu',
      content: 'plus4u5-app-top-left-content',
      buttonGroup: 'plus4u5-app-top-left-button-group'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getPropsToPass: function _getPropsToPass() {
    return UU5.Common.Tools.merge(this.getMainPropsToPass(), { content: null });
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    //TODO - ContainerCollection
    return this.getNestingLevel() ? _react2.default.createElement(
      UU5.Bricks.Div,
      this._getPropsToPass(),
      _react2.default.createElement(UU5.Bricks.Page.MenuButton, { menu: "left", colorSchema: "custom", className: this.getClassName('leftMenu') }),
      _react2.default.createElement(UU5.Bricks.Span, { className: this.getClassName('content'), content: this.props.content || this.props.children })
    ) : null;
  }
} //@@viewOff:render

);

exports.default = Top;
