// uu5g04/dist-node/bricks/tabs-item.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

var _bricksNs = require('./bricks-ns.js');

var _bricksNs2 = _interopRequireDefault(_bricksNs);

require('./tabs-item.less');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _createReactClass2.default)({
  displayName: 'tabs-item',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.PureRenderMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin, UU5.Common.SectionMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: _bricksNs2.default.name("Tabs.Item"),
    nestingLevelList: UU5.Environment.getNestingLevelList('bigBoxCollection', 'smallBox'),
    classNames: {
      main: _bricksNs2.default.css("tabs-item", "tabs-item-pane"),
      hidden: _bricksNs2.default.css("tabs-item-hidden"),
      visible: _bricksNs2.default.css("tabs-item-visible"),
      fade: _bricksNs2.default.css("tabs-item-fade ")
    },
    opt: {
      nestingLevelWrapper: true
    },
    errors: {
      invalidParent: 'Parent of this component is not Tabs.'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    _fade: _propTypes2.default.bool,
    _active: _propTypes2.default.bool
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      _fade: false,
      _active: false
    };
  },
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  getInitialState: function getInitialState() {
    return {
      visibility: this.props._active
    };
  },

  componentWillMount: function componentWillMount() {
    var parent = this.getParent();

    if (parent) {
      while (parent.getOpt("parentWrapper")) {
        parent = parent.getParent();
      }
    }

    if (!(parent && parent.isTabs)) {
      this.showError("invalidParent");
    }
  },

  componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
    if (!nextProps._active && nextProps._active != this.props._active) {
      this.setState({ visibility: false });
    }
  },

  componentDidUpdate: function componentDidUpdate(prevProps, prevState) {
    var _this = this;

    if (this.props._fade && this.props._active && prevProps._active != this.props._active) {
      setTimeout(function () {
        return _this.setAsyncState({ visibility: true });
      }, 5);
    } else if (this.props._active && prevProps._active != this.props._active) {
      this.setState({ visibility: true });
    }
  },

  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var mainAttrs = this.getMainAttrs();

    if (this.props._active) {
      if (this.state.visibility) {
        mainAttrs.className += ' ' + this.getClassName().visible;
      } else {
        mainAttrs.className += ' ' + this.getClassName().hidden;
      }
    }
    if (this.props._fade) {
      mainAttrs.className += ' ' + this.getClassName().fade;
    }

    return this.getNestingLevel() ? _react2.default.createElement(
      'div',
      mainAttrs,
      this.getChildren(),
      this.getDisabledCover()
    ) : null;
  }

  //@@viewOff:render
});
