// uu_plus4u5g01/dist-node/uubmldraw/image/image-legacy.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ImageLegacy = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _uu_plus4u5g = require('uu_plus4u5g01');

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

require('./image-legacy.css');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ImageLegacy = exports.ImageLegacy = (0, _createReactClass2.default)({
  displayName: 'ImageLegacy',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.UuBmlDraw.ImageLegacy_',
    classNames: {
      main: 'plus4u5-uubmldraw-image plus4u5-uubmldraw-image-legacy',
      modal: 'plus4u5-uubmldraw-image-modal-legacy',
      iframe: 'plus4u5-uubmldraw-image-iframe-legacy'
    },
    defaults: {
      width: 250,
      height: 150,
      contentUrl: "https://widget.plus4u.net/%s/uu-uubmldraw/getContent?referrerUri=%s&uuid=%s&uuAT=Bearer %s&readOnly=%s",
      sheetUri: "ues:VPH-BT[29537617]:UUAPPKNH/PORTAL[44191581396036005]:50946987294721227[50946987294721227]"
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    sheetUri: _propTypes2.default.string.isRequired,
    code: _propTypes2.default.string.isRequired,
    readOnly: _propTypes2.default.bool,
    inProgress: _propTypes2.default.bool,
    userIdentity: _propTypes2.default.string,
    userName: _propTypes2.default.string
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      readOnly: false,
      inProgress: false,
      userIdentity: null,
      userName: null
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  componentDidMount: function componentDidMount() {
    window.uuBMLDraw = window.uuBMLDraw || {};
    uuBMLDraw[this.props.code] = uuBMLDraw[this.props.code] || {};
    uuBMLDraw.initStack = uuBMLDraw.initStack || [];
    UU5.Environment.EventListener.addWindowEvent('message', this.getId(), this._receiveMessage);
  },
  componentWillUnmount: function componentWillUnmount() {
    UU5.Environment.EventListener.removeWindowEvent('message', this.getId());
    delete uuBMLDraw[this.props.code];
  },

  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  showImage: function showImage() {
    if (uuBMLDraw[this.props.code] && uuBMLDraw[this.props.code].image) {
      this._modal.open({
        content: _react2.default.createElement(UU5.Bricks.Image, { src: uuBMLDraw[this.props.code].image })
      });
    }
  },

  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getIframeProps: function _getIframeProps() {
    var _this = this;

    return {
      className: this.getClassName().iframe,
      src: this._buildSrc(),
      inline: true,
      resize: true,
      ref_: function ref_(iframe) {
        return _this._iframe = iframe;
      },
      iframeAttrs: {
        name: this.getId()
      }
    };
  },
  _buildSrc: function _buildSrc() {
    var sheetUri = this.props.sheetUri || this.getDefault().sheetUri;
    return UU5.Common.Tools.formatString(this.getDefault().contentUrl, [this._getTerritoryCode(sheetUri), sheetUri, this.props.code, Plus4U5.Environment.getToken(), this.props.readOnly]);
  },
  _getTerritoryCode: function _getTerritoryCode(uri) {
    var matcher = uri.match(/:([A-Z0-9-]+)/);
    return matcher[1];
  },
  _receiveMessage: function _receiveMessage(event) {
    var message = event.data;

    var widgetData = uuBMLDraw[this.props.code] || {};

    if (message.name === this.getId()) {
      switch (message.type) {
        case 'setDimensions':
        case 'restoreDimensions':
          this._iframe.setSize(message.width, message.height + 10);
          break;
        case 'setImage':
          var img = message.base64;
          widgetData.image = img;
          this._iframe.setSize(message.width, message.height + 10);
          break;
        case 'showImages':
          this.showImage();
          break;
        default:
          break;
      }
    }
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var _this2 = this;

    return _react2.default.createElement(
      'div',
      this.getMainAttrs(),
      _react2.default.createElement(UU5.Bricks.Modal, { parent: this, header: ' ', ref_: function ref_(modal) {
          return _this2._modal = modal;
        }, className: this.getClassName().modal }),
      _react2.default.createElement(UU5.Bricks.Iframe, this._getIframeProps())
    );
  }
  //@@viewOff:render

});

exports.default = ImageLegacy;
