// uu_plus4u5g01/dist-node/uubmldraw/image/image-loader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ImageLoader = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _slicedToArray = function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _uu_plus4u5g = require('uu_plus4u5g01');

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

var _calls = require('../calls.js');

var _calls2 = _interopRequireDefault(_calls);

require('./image.css');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ImageLoader = exports.ImageLoader = (0, _createReactClass2.default)({
  displayName: 'ImageLoader',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.LsiMixin, UU5.Common.LoadMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.UuBmlDraw.ImageLoader',
    classNames: {
      main: 'plus4u5-uubmldraw-image',
      modal: 'plus4u5-uubmldraw-image-modal',
      button: 'plus4u5-uubmldraw-image-button uu5-elevation-2',
      inProgress: 'plus4u5-uubmldraw-image-inprogress',
      img: 'plus4u5-uubmldraw-image-img uu5-elevation-0',
      imageWrapper: "plus4u5-uubmldraw-image-image-wrapper"
    },
    defaults: {
      width: 250,
      height: 150,
      locked: 'mdi-lock-outline',
      editable: 'mdi-lead-pencil',
      inProgress: Plus4U5.Environment.basePath + 'uubmldraw/assets/in-progress.png',
      placeholder: Plus4U5.Environment.basePath + 'uubmldraw/assets/placeholder.png'
    },
    calls: {
      onLoad: "loadDiagram",
      lock: "lock"
    },
    lsi: {
      tooltip: {
        cs: "Editovat",
        en: "Edit"
      },
      lockedAlert: {
        cs: "Editor je uzamčen uživatelem %s (%s)",
        en: "The editor is currently locked by %s (%s)"
      }
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    tidAwid: _propTypes2.default.string.isRequired,
    code: _propTypes2.default.string.isRequired,
    readOnly: _propTypes2.default.bool
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      tidAwid: null,
      readOnly: false
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  componentWillMount: function componentWillMount() {
    this.setCalls(_calls2.default);
  },
  componentWillUnmount: function componentWillUnmount() {
    UU5.Environment.EventListener.removeWindowEvent('message', this.getId());
    return this;
  },

  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  edit: function edit() {
    var _this = this;

    if (this.editor) {
      this.editor.focus();
    } else {
      this.getCall('lock')({
        tidAwid: this.props.tidAwid,
        data: {
          code: this.props.code
        },

        done: function done(dtoOut) {
          _this._handleLock(dtoOut);
          _this.setState({ dtoOut: dtoOut });
        },

        fail: function fail(failDtoOut) {
          console.error(failDtoOut);
        }
      });
    }
    return this;
  },
  showAlert: function showAlert() {
    var dtoOut = this.getDtoOut();
    var lock = dtoOut.lock || {};
    this._alertBus.addAlert({
      content: this.getLsiComponent('lockedAlert', null, [lock.author, lock.uuIdentity]),
      colorSchema: 'warning'
    });
    return this;
  },
  showImage: function showImage() {
    this._modal.open({
      content: _react2.default.createElement(UU5.Bricks.Image, { src: this._buildImgSrc() })
    });
    return this;
  },

  //@@viewOff:interface

  //@@viewOn:overridingMethods
  onLoadError_: function onLoadError_(dtoOut) {
    this.setState({ loadFeedback: "error" });
  },
  getOnLoadData_: function getOnLoadData_() {
    return {
      code: this.props.code,
      tidAwid: this.props.tidAwid
    };
  },


  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _receiveMessage: function _receiveMessage(event) {
    // ((event.origin !== "http://localhost:6221"))
    if (event.origin !== "https://uuos9.plus4u.net") return event;
    try {
      var _event$data$split = event.data.split("."),
          _event$data$split2 = _slicedToArray(_event$data$split, 2),
          code = _event$data$split2[0],
          message = _event$data$split2[1];

      if (code === this.props.code) {
        switch (message) {
          case "unlock":
            this.editor = null;
            this.reload();
            UU5.Environment.EventListener.removeWindowEvent('message', this.getId());
        }
      }
    } catch (e) {
      console.error(e);
    }
  },
  _handleButtonClick: function _handleButtonClick() {
    this.edit();
  },
  _handleLock: function _handleLock(dtoOut) {
    if (dtoOut.lock) {
      var url = _calls2.default.getCommandUri('editor', this.props.tidAwid, {
        code: this.props.code,
        inProgress: (!!dtoOut.inProgress).toString(),
        image: (!!dtoOut.image).toString()
      }).toString();
      this.editor = window.open(url);
      UU5.Environment.EventListener.addWindowEvent('message', this.getId(), this._receiveMessage);
    } else {
      this.showAlert();
    }
  },
  _getButtonTooltip: function _getButtonTooltip() {
    var lock = this.getDtoOut().lock || {};
    if (lock.uuIdentity) {
      return lock.author + '\n' + lock.uuIdentity;
    }
  },
  _getCorrectIcon: function _getCorrectIcon() {
    var lock = this.getDtoOut().lock;
    var def = this.getDefault();
    return (!!lock && Object.keys(lock).length) > 0 ? def.locked : def.editable;
  },
  _buildButtonAttrs: function _buildButtonAttrs() {
    var _this2 = this;

    var attrs = {
      className: this.getClassName().button,
      hidden: true,
      size: "xl",
      ref_: function ref_(button) {
        return _this2._button = button;
      },
      onClick: function onClick(button, e) {
        e.stopPropagation();
        _this2.edit();
      }
    };
    var tooltip = this._getButtonTooltip();
    tooltip && (attrs.tooltip = tooltip);
    return attrs;
  },
  _buildModalAttrs: function _buildModalAttrs() {
    var _this3 = this;

    return {
      header: this.getDtoOut().name || " ",
      className: this.getClassName().modal,
      ref_: function ref_(modal) {
        _this3._modal = modal;
      }
    };
  },
  _buildImgSrc: function _buildImgSrc() {
    return this.getDtoOut().imgUrl || this.getDefault().placeholder;
  },
  _buildImageAttrs: function _buildImageAttrs() {
    var _this4 = this;

    return {
      src: this._buildImgSrc(),
      className: this.getClassName().img,
      ref_: function ref_(image) {
        return _this4._image = image;
      }
    };
  },
  _onMouseOverHandler: function _onMouseOverHandler() {
    !this.getDtoOut().readOnly && this._button.show();
  },
  _onMouseOutHandler: function _onMouseOutHandler() {
    this._button.hide();
  },
  _buildInProgressBar: function _buildInProgressBar() {
    if (this.getDtoOut().inProgress) {
      return _react2.default.createElement(UU5.Bricks.Image, this._buildInProgressProps());
    }
  },
  _buildInProgressProps: function _buildInProgressProps() {
    return {
      className: this.getClassName().inProgress,
      src: this.getDefault().inProgress
    };
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var _this5 = this;

    return this.getLoadFeedbackChildren(function () {
      return _react2.default.createElement(
        'div',
        _extends({}, _this5.getMainAttrs(), { onMouseOver: function onMouseOver() {
            return _this5._onMouseOverHandler();
          }, onMouseOut: function onMouseOut() {
            return _this5._onMouseOutHandler();
          } }),
        _react2.default.createElement(UU5.Bricks.Modal, _this5._buildModalAttrs()),
        _react2.default.createElement(UU5.Bricks.AlertBus, { ref_: function ref_(alert) {
            return _this5._alertBus = alert;
          } }),
        _react2.default.createElement(
          'div',
          { onClick: function onClick() {
              return _this5.showImage();
            }, className: _this5.getClassName().imageWrapper },
          _this5._buildInProgressBar(),
          _react2.default.createElement(UU5.Bricks.Image, _this5._buildImageAttrs()),
          _react2.default.createElement(
            UU5.Bricks.Button,
            _this5._buildButtonAttrs(),
            _react2.default.createElement(UU5.Bricks.Icon, { icon: _this5._getCorrectIcon() })
          )
        )
      );
    });
  }
  //@@viewOff:render

});

exports.default = ImageLoader;
