// uu_plus4u5g01/dist-node/uubmldraw/image/image.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Image = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _imageLoader = require('./image-loader.js');

var _imageLoader2 = _interopRequireDefault(_imageLoader);

var _imageLegacy = require('./image-legacy.js');

var _imageLegacy2 = _interopRequireDefault(_imageLegacy);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Image = exports.Image = (0, _createReactClass2.default)({
  displayName: 'Image',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.IdentityMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.UuBmlDraw.Image',
    lsi: {
      codeError: {
        cs: "Chybí povinná vlastnost 'code'!",
        en: "Required property 'code' is missing!"
      }
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    code: _propTypes2.default.string.isRequired
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getCorrectChild: function _getCorrectChild() {
    if (!this.props.code) return _react2.default.createElement(UU5.Bricks.Error, { content: this.getLsiComponent("codeError") });

    var identity = this.getIdentity();
    var props = Object.assign({}, this.props, { userIdentity: identity.uuIdentity, userName: identity.name });
    return this.props.tidAwid ? _react2.default.createElement(_imageLoader2.default, props) : _react2.default.createElement(_imageLegacy2.default, props);
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var _this = this;

    return this.getAuthenticatedChild(function () {
      return _this._getCorrectChild();
    }, { silent: true, inline: true });
  }
  //@@viewOff:render

});

exports.default = Image;
