// uu_plus4u5g01/dist-node/bricks/cmd-viewer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CmdViewer = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _calls = require('./calls.js');

var _calls2 = _interopRequireDefault(_calls);

require('./cmd-viewer.less');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CmdViewer = exports.CmdViewer = (0, _createReactClass2.default)({
  displayName: 'CmdViewer',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.LoadMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.CmdViewer',
    classNames: {
      main: 'plus4u5-bricks-cmd-viewer'
    },
    calls: {
      onLoad: "loadCmd"
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    url: _propTypes2.default.string.isRequired,
    params: _propTypes2.default.object,
    theme: _propTypes2.default.string,
    minRows: _propTypes2.default.number,
    maxRows: _propTypes2.default.number,
    rows: _propTypes2.default.number,
    hideMarks: _propTypes2.default.bool,
    loadMethod: _propTypes2.default.oneOf(['get', 'post'])
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      url: null,
      params: null,
      theme: "chrome",
      minRows: null,
      maxRows: 50,
      rows: null,
      hideMarks: false,
      loadMethod: 'get'
    };
  },
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  componentWillMount: function componentWillMount() {
    this.setCalls(_calls2.default);
  },

  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  getOnLoadData_: function getOnLoadData_(props) {
    return {
      url: props.url,
      params: props.params,
      loadMethod: props.loadMethod.toLowerCase()
    };
  },

  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var _this = this;

    return this.getLoadFeedbackChildren(function (dtoOut) {
      return UU5.Common.Tools.findComponent('UU5.CodeKit.JsonEditor', UU5.Common.Tools.merge({}, _this.getMainPropsToPass(), {
        value: dtoOut,
        format: "pretty",
        readOnly: true,
        theme: _this.props.theme,
        minRows: _this.props.minRows,
        maxRows: _this.props.rows ? null : _this.props.maxRows,
        rows: _this.props.rows,
        ref_: function ref_(editor) {
          if (!_this.props.hideMarks) {
            // TODO not really good
            setTimeout(function () {
              var aceEditor = editor._textInput.editor;
              aceEditor.getSession().setFoldStyle("markbegin");
            }, 1000);
          }
        }
      }));
    });
  }
  //@@viewOff:render
});

exports.default = CmdViewer;
