// uu_plus4u5g01/dist-node/bricks/error.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Error = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _uu_plus4u5g = require('uu_plus4u5g01');

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Error = exports.Error = (0, _createReactClass2.default)({
  displayName: 'Error',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.ContentMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.Error',
    classNames: {
      main: 'plus4u5-bricks-error'
    },
    lsi: function lsi() {
      return Plus4U5.Environment.Lsi.Bricks.error;
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    errorData: _propTypes2.default.object,
    error: _propTypes2.default.shape({
      stack: _propTypes2.default.string,
      name: _propTypes2.default.string,
      message: _propTypes2.default.string
    }),
    silent: _propTypes2.default.bool,
    inline: _propTypes2.default.bool,
    errorDataSpace: _propTypes2.default.number // TODO: set deprecated

  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      errorData: null,
      error: null,
      silent: false,
      inline: false,
      errorDataSpace: null
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var props = this.getMainPropsToPass();
    var children = void 0;

    if (!this.props.children && !this.getContent()) {
      props.content = this.getLsiComponent('message');
      if (this.props.inline) {
        props.content = [_react2.default.createElement(
          'span',
          { key: 'message', style: { marginRight: "4px" } },
          this.getLsiComponent('messageFull')
        ), _react2.default.createElement(Plus4U5.Bricks.LinkHelpdesk, { key: 'LinkHelpDesk' })];
      }
    } else if (this.props.children) {
      children = _react2.default.Children.toArray(this.props.children);
    }

    return _react2.default.createElement(
      UU5.Bricks.Error,
      _extends({}, props, { silent: this.props.silent, errorData: this.props.errorData, inline: this.props.inline, error: this.props.error, moreInfo: true, errorInfo: _react2.default.createElement(
          'div',
          null,
          this.getLsiComponent('content'),
          ' ',
          _react2.default.createElement(Plus4U5.Bricks.LinkHelpdesk, null)
        ) }),
      children
    );
  }
  //@@viewOff:render
});

exports.default = Error;
