// uu_plus4u5g01/dist-node/bricks/iframe.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Iframe = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _calls = require('./calls');

var _calls2 = _interopRequireDefault(_calls);

require('./iframe.less');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Iframe = exports.Iframe = (0, _createReactClass2.default)({
  displayName: 'Iframe',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.LoadMixin, UU5.Common.IdentityMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.Iframe',
    classNames: {
      main: 'plus4u5-bricks-iframe'
    },
    calls: {
      onLoad: 'getData'
    },
    lsi: function lsi() {
      return Plus4U5.Environment.Lsi.Bricks.iframe;
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    src: _propTypes2.default.string,
    resize: _propTypes2.default.bool,
    height: _propTypes2.default.string,
    syncTimeout: _propTypes2.default.number,
    inline: _propTypes2.default.bool,
    iframeAttrs: _propTypes2.default.object
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      src: null,
      resize: undefined, // same as UU5.Bricks.Iframe
      height: undefined, // same as UU5.Bricks.Iframe
      syncTimeout: undefined, // same as UU5.Bricks.Iframe
      inline: undefined, // same as UU5.Bricks.Iframe
      iframeAttrs: undefined // same as UU5.Bricks.Iframe
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  componentWillMount: function componentWillMount() {
    this.setCalls(_calls2.default);
  },

  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  getOnLoadData_: function getOnLoadData_(props) {
    return {
      src: props.src
    };
  },

  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var child = void 0;
    switch (this.getLoadFeedback()) {
      case "loading":
        child = _react2.default.createElement(UU5.Bricks.Loading, this.getMainPropsToPass());
        break;
      case "ready":
        child = _react2.default.createElement(UU5.Bricks.Iframe, _extends({}, this.getMainAttrs(), { src: this.getDtoOut(), resize: this.props.resize, height: this.props.height, syncTimeout: this.props.syncTimeout, inline: this.props.inline, iframeAttrs: this.props.iframeAttrs }));
        break;
      default:
        child = _react2.default.createElement(
          UU5.Bricks.Error,
          this.getMainPropsToPass(),
          this.getLsiComponent('notFound')
        );
    }
    return child;
  }
  //@@viewOff:render

});

exports.default = Iframe;
