// uu_plus4u5g01/dist-node/bricks/image.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Image = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _uu_plus4u5g = require('uu_plus4u5g01');

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

require('./image.less');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Image = exports.Image = (0, _createReactClass2.default)({
  displayName: 'Image',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.Image',
    classNames: {
      main: 'plus4u5-bricks-image'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    src: _propTypes2.default.string,
    type: _propTypes2.default.oneOf(['rounded', 'circle', 'thumbnail'])
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      src: null,
      type: null
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getSrc: function _getSrc() {
    var src = this.props.src;
    var token = Plus4U5.Environment.getToken();

    var divider = /\?/.test(src) ? '&' : '?';
    if (token) {
      var url = src + divider + 'access_token=%s';
      src = UU5.Common.Tools.formatString(url, token);
    }

    return src;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return this.props.src ? _react2.default.createElement(UU5.Bricks.Image, _extends({}, this.props, this.getMainPropsToPass(), { src: this._getSrc() })) : null;
  }
  //@@viewOff:render

});

exports.default = Image;
