// uu_plus4u5g01/dist-node/bricks/link.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Link = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _uu_plus4u5g = require('uu_plus4u5g01');

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

require('./link.less');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Link = exports.Link = (0, _createReactClass2.default)({
  displayName: 'Link',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.Link',
    classNames: {
      main: 'plus4u5-bricks-link'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    href: _propTypes2.default.string,
    onClick: _propTypes2.default.func,
    onCtrlClick: _propTypes2.default.func,
    onWheelClick: _propTypes2.default.func,
    smoothScroll: _propTypes2.default.number,
    offset: _propTypes2.default.number,
    target: _propTypes2.default.string
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getHref: function _getHref() {
    var href = this.props.href;
    var token = Plus4U5.Environment.getToken();

    var divider = /\?/.test(href) ? '&' : '?';
    if (token) {
      var url = href + divider + 'access_token=%s';
      href = UU5.Common.Tools.formatString(url, token);
    }

    return href;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    return _react2.default.createElement(UU5.Bricks.Link, _extends({}, this.getMainPropsToPass(), this.props, { href: this._getHref() }));
  }
  //@@viewOff:render

});

exports.default = Link;
