// uu_plus4u5g01/dist-node/bricks/plus4u-link.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Plus4ULink = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _uu_plus4u5g = require('uu_plus4u5g01');

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

require('./plus4u-link.less');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Plus4ULink = exports.Plus4ULink = (0, _createReactClass2.default)({
  displayName: 'Plus4ULink',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.Plus4ULink',
    classNames: {
      main: 'plus4u5-bricks-plus4u-link'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    uri: _propTypes2.default.string,
    target: _propTypes2.default.string
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      uri: null,
      target: "_blank"
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _onClick: function _onClick(e, target) {
    window.open(Plus4U5.Common.Tools.getLegacyUrl(this.props.uri), target || this.props.target);
    e.preventDefault();
    return this;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var _this = this;

    return _react2.default.createElement(
      UU5.Bricks.Link,
      _extends({}, this.getMainPropsToPass(), { href: Plus4U5.Common.Tools.getLegacyUrl(this.props.uri), onClick: function onClick(component, e) {
          return _this._onClick(e);
        }, onCtrlClick: function onCtrlClick(component, e) {
          return _this._onClick(e, "_blank");
        }, onWheelClick: function onWheelClick(component, e) {
          return _this._onClick(e, "_blank");
        }, content: this.props.content }),
      this.props.children
    );
  }
  //@@viewOff:render

});

exports.default = Plus4ULink;
