// uu_plus4u5g01/dist-node/bricks/user-photo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UserPhoto = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require('create-react-class');

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require('uu5g04');

var UU5 = _interopRequireWildcard(_uu5g);

require('uu5g04-bricks');

var _uu_plus4u5g = require('uu_plus4u5g01');

var Plus4U5 = _interopRequireWildcard(_uu_plus4u5g);

var _uu_appg01_core = require('uu_appg01_core');

require('./user-photo.less');

var _richImage = require('./rich-image.js');

var _richImage2 = _interopRequireDefault(_richImage);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var UserPhoto = exports.UserPhoto = (0, _createReactClass2.default)({
  displayName: 'UserPhoto',


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.IdentityMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.UserPhoto',
    classNames: {
      main: 'plus4u5-bricks-user-photo'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    uuIdentity: _propTypes2.default.string,
    src: _propTypes2.default.string,
    type: _propTypes2.default.oneOf(['rounded', 'circle', 'thumbnail']),
    width: _propTypes2.default.number,
    shadow: _propTypes2.default.bool
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      uuIdentity: null,
      src: null,
      type: null,
      width: 50
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _getUserPhoto: function _getUserPhoto(uuIdentity) {
    var uri = void 0;

    if (this.isAuthenticated()) {
      var token = Plus4U5.Environment.getToken();

      if (token) {
        if (this.props.src) {
          try {
            var url = _uu_appg01_core.Uri.UriBuilder.parse(this.props.src);
            url.setParameter('access_token', token);
            uri = url.toUri().toString();
          } catch (e) {
            uri = this.props.src;
          }
        } else {
          var _url = 'https://cmd.plus4u.net/PLUS4U-BT/uu-plus4uppl/PersonalCard/getPhotoByUuId/exec?uuUri=ues:PLUS4U-BT:PLUS4U-BT&uuIdentity=%s&uuAT=Bearer%20%s';
          uri = UU5.Common.Tools.formatString(_url, [uuIdentity || this.getIdentity().uuIdentity, token]);
        }
      }
    }

    uri = uri || this.props.src || Plus4U5.Environment.basePath + 'core/assets/img/anonymous.png';

    return uri;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function render() {
    var props = this.getMainPropsToPass();
    props.mainAttrs = props.mainAttrs || {};
    props.mainAttrs.style = props.mainAttrs.style || {};
    props.mainAttrs.style.width = this.props.width;

    return _react2.default.createElement(_richImage2.default, _extends({}, props, { name: this.props.uuIdentity, src: this._getUserPhoto(this.props.uuIdentity), shape: this.props.type, effect: 'none', shadow: this.props.shadow }));
  }
  //@@viewOff:render

});

exports.default = UserPhoto;
