// uu_plus4u5g01/dist-node/bricks/uu5string-update.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Uu5StringUpdate = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _createReactClass = require("create-react-class");

var _createReactClass2 = _interopRequireDefault(_createReactClass);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uu5g = require("uu5g04");

var UU5 = _interopRequireWildcard(_uu5g);

require("uu5g04-bricks");

require("uu5g04-forms");

require("./uu5string-update.css");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Uu5StringUpdate = exports.Uu5StringUpdate = (0, _createReactClass2.default)({
  displayName: "Uu5StringUpdate",


  //@@viewOn:mixins
  mixins: [UU5.Common.BaseMixin, UU5.Common.ElementaryMixin, UU5.Common.NestingLevelMixin],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'Plus4U5.Bricks.Uu5StringUpdate',
    nestingLevelMixin: 'boxCollection',
    classNames: {
      main: 'plus4u5-bricks-uu5string-update'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  propTypes: {
    data: _propTypes2.default.string,
    save: _propTypes2.default.func,
    saveDone: _propTypes2.default.func,
    saveFail: _propTypes2.default.func,
    cancel: _propTypes2.default.func,
    validationMessage: _propTypes2.default.any,
    saveMessage: _propTypes2.default.any,
    saveFailMessage: _propTypes2.default.any,
    format: _propTypes2.default.string
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  getDefaultProps: function getDefaultProps() {
    return {
      data: null,
      save: null,
      saveDone: null,
      saveFail: null,
      cancel: null,
      validationMessage: null,
      saveMessage: null,
      saveFailMessage: null,
      format: "pretty"
    };
  },

  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  _validate: function _validate() {
    var input = this._uu5StringEdit;
    var value = input.getValue();
    var validFlag = true;
    this._formData = null;
    try {
      this._formData = value; //TODO Validate uu5String
    } catch (err) {
      validFlag = false;
      this._form.getAlertBus().setAlert({
        content: this.props.validationMessage || 'uu5String is not correct!',
        colorSchema: 'danger'
      });
    }

    return validFlag;
  },
  _save: function _save(opt, justSave) {
    if (this._validate()) {
      var dtoIn = {
        done: function done(dtoOut) {
          return opt.component.saveDone(dtoOut, justSave);
        },
        fail: function fail(dtoOut) {
          return opt.component.saveFail(dtoOut);
        },
        data: this._formData
      };
      typeof this.props.save === 'function' && this.props.save(dtoIn);
      this._form.getAlertBus().setAlert({
        content: this.props.saveMessage || _react2.default.createElement(UU5.Bricks.Loading, { style: { minWidth: '100px' } }),
        colorSchema: 'info'
      });
    }
  },
  _saveDone: function _saveDone(dtoOut, justSave) {
    this._form.getAlertBus().clearAlerts();
    typeof this.props.saveDone === 'function' && this.props.saveDone(dtoOut, justSave);
  },
  _saveFail: function _saveFail(dtoErr) {
    typeof this.props.saveFail === 'function' && this.props.saveFail(dtoErr);
    this._form.getAlertBus().setAlert({ content: this.props.saveFailMessage || "Save failed!", colorSchema: 'danger' });
  },
  _reset: function _reset() {
    this._form.getAlertBus().clearAlerts();
    this._uu5StringEdit.reset();
  },
  _cancel: function _cancel() {
    this._form.getAlertBus().clearAlerts();
    typeof this.props.cancel === 'function' && this.props.cancel();
  },
  _getMainProps: function _getMainProps() {
    var _this = this;

    var props = this.getMainPropsToPass();
    props.onSave = function (opt) {
      return _this._save(opt);
    };
    props.onSaveByKey = function (opt) {
      return _this._save(opt, true);
    };
    props.onSaveDone = function (opt) {
      return _this._saveDone(opt);
    };
    props.onSaveFail = function (opt) {
      return _this._saveFail(opt);
    };
    props.onCancel = this._cancel;
    props.onReset = this._reset;
    props.onValidate = this._validate;
    props.ref_ = function (form) {
      return _this._form = form;
    };
    return props;
  },

  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render

  render: function render() {
    var _this2 = this;

    var uu5StringEditor = UU5.Common.Tools.findComponent('UU5.CodeKit.Uu5StringEditor', {
      ref_: function ref_(component) {
        return _this2._uu5StringEdit = component;
      },
      value: this.props.data,
      rows: 20,
      focus: true
    });

    return _react2.default.createElement(
      UU5.Forms.Form,
      this._getMainProps(),
      uu5StringEditor,
      _react2.default.createElement(UU5.Forms.Controls, null)
    );
  }
  //@@viewOff:render

});

exports.default = Uu5StringUpdate;
